casper/src/main/rholang/BasicWallet.rho

```diff
@@ -0,0 +1,67 @@
+//scalapackage coop.rchain.rholang.wallet
+
+//assumes standard purse API from MakeMint
+contract @"BasicWallet"(@purse, @algorithm, @pk, return) = {
+  new algorithmKnown, cryptoVerifyCh in {
+    match algorithm {
+      "ed25519" => {
+        algorithmKnown!(true) | cryptoVerifyCh!("ed25519Verify")
+      }
+      _ => {
+        algorithmKnown!(false)
+      }
+    } |
+    for(@ak <- algorithmKnown) {
+      match ak {
+        false => { return!([]) }
+        true => {
+          for (cryptoVerify <- cryptoVerifyCh) {
+            new basicWallet, verify, nonceCh in {
+              nonceCh!(-1) |
+              contract verify(@nonce, @sig, return) = {
+                for(@prevNonce <- nonceCh) {
+                  match nonce == (prevNonce + 1) {
+                    true => { 
+                      new result in {
+                        cryptoVerify!(nonce.toByteArray(), sig.hexToBytes(), pk.hexToBytes(), *result) |
+                        for(@r <- result) {
+                          match r {
+                            true => { nonceCh!(nonce) | return!(true) }
+                            _    => { nonceCh!(prevNonce) | return!(false) }
+                          }
+                        }
+                      }
+                    }
+                    _ => { nonceCh!(prevNonce) | return!(false) }
+                  }
+                }
+              } |
+              contract @[*basicWallet, "getNonce"](return) = {
+                for(@nonce <- nonceCh) {
+                  nonceCh!(nonce) | return!(nonce)
+                }
+              } |
+              contract @[*basicWallet, "getBalance"](return) = {
+                @[purse, "getBalance"]!(*return)
+              } |
+              contract @[*basicWallet, "deposit"](@amount, @src, success) = {
+                @[purse, "deposit"]!(amount, src, *success)
+              } |
+              contract @[*basicWallet, "withdraw"](@amount, @nonce, @sig, return) = {
```

Comment:
The only thing a wallet should do is verify the signature on a message and then forward it to the underlying purse.  If you think there should be a withdraw method, please put it on the purse itself.